tests/rbd/rbd_utils.py

```diff
@@ -141,11 +142,18 @@ def create_image(self, pool_name, image_name, size, **kw):
 
     def set_ec_profile(self, profile):
         self.exec_cmd(cmd="ceph osd erasure-code-profile rm {}".format(profile))
-        self.exec_cmd(
-            cmd="ceph osd erasure-code-profile set {} k={} m={}".format(
-                profile, self.k_m[0], self.k_m[2]
+        if self.failure_domain == "osd":
+            self.exec_cmd(
+                cmd="ceph osd erasure-code-profile set {} crush-failure-domain=osd k={} m={}".format(
```

Comment:
We don't need to revert it back as we are setting this for EC profile related to particular EC pool. And it will not be set for new EC profile created for other test case.